packages/alchemy/src/client/types.ts

```diff
@@ -16,30 +21,52 @@ export type AlchemyRpcSchema = [
       {
         policyId: string;
         entryPoint: Address;
-        userOperation: UserOperationRequest;
+        userOperation: UserOperationRequest<EntryPointVersion>;
       }
     ];
-    ReturnType: { paymasterAndData: Hex };
+    ReturnType:
+      | {
+          paymasterAndData: UserOperationRequest<"0.6.0">["paymasterAndData"];
+        }
+      | Pick<
+          UserOperationRequest<"0.7.0">,
+          | "paymaster"
+          | "paymasterData"
+          | "paymasterPostOpGasLimit"
+          | "paymasterVerificationGasLimit"
+        >;
```

Comment:
left a comment about this somewhere in the last review, but let's make this a type alias and use it here and below and anywhere else we need a union type for the `PaymasterAndDataFields`